pkg/operators/wasm/maps.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"context"
+
+	"github.com/cilium/ebpf"
+	"github.com/tetratelabs/wazero"
+	wapi "github.com/tetratelabs/wazero/api"
+)
+
+func (i *wasmOperatorInstance) addMapFuncs(env wazero.HostModuleBuilder) {
+	exportFunction(env, "getMap", i.getMap,
+		[]wapi.ValueType{wapi.ValueTypeI64}, // MapName
+		[]wapi.ValueType{wapi.ValueTypeI32}, // Map
```

Comment:
That sounds good to me.

Later we can add another function `getMapByID(uint32) uint32` so custom gadgets could pick up maps that were created externally (e.g. by bpftrace or another third-party tool). This was a use case from discussions at the All Systems Go conference.